### PR TITLE
Arcade Machine Powercreep

### PR DESCRIPTION
**Pull request by BurgerLUA**
Christmas is ending soon, and I enjoy getting some of the items from the gifts. This update adds most items to the random arcade machine loot as well as tweaks the RNG for some of the rewards. No longer will you get a shitton of toy mechs. Check the arcade machine file for the weights and items in the arcade machine now. Higher values means higher chance for that item to spawn.

Some descriptions of toys were cleaned up a little, and the toy sword now gets a random color scheme upon initialization.

### CODE REVIEW

File: code/game/machinery/computer/arcade.dm [hunk 1]
```diff
@@ -24,11 +23,33 @@
...
+							/obj/item/weapon/bikehorn						= 11,
+							/obj/item/weapon/lipstick/random				= 11,
+							/obj/item/clothing/mask/fakemoustache			= 11,
+							/obj/item/clothing/mask/gas/clown_hat			= 11,
+							/obj/item/clothing/mask/gas/mime				= 11,
+							/obj/item/weapon/gun/energy/wand/toy			= 11,
+							/obj/item/weapon/gun/energy/mousegun			= 11,
+							/obj/item/weapon/stamp/clown					= 11,
+							/obj/item/device/binoculars						= 11,
+							/obj/item/weapon/autochisel						= 11,
```
Comment (Pacmandevil) [inline code/game/machinery/computer/arcade.dm]: This should stay behind mining,

File: code/game/machinery/computer/arcade.dm [hunk 2]
```diff
@@ -24,11 +23,33 @@
...
+							/obj/item/weapon/storage/belt/champion			= 11,
+							/obj/item/weapon/pen/invisible					= 11,
+							/obj/item/weapon/grenade/smokebomb				= 1,
+							/obj/item/weapon/bikehorn						= 11,
+							/obj/item/weapon/lipstick/random				= 11,
+							/obj/item/clothing/mask/fakemoustache			= 11,
+							/obj/item/clothing/mask/gas/clown_hat			= 11,
+							/obj/item/clothing/mask/gas/mime				= 11,
+							/obj/item/weapon/gun/energy/wand/toy			= 11,
+							/obj/item/weapon/gun/energy/mousegun			= 11,
```
Comment (Pacmandevil) [inline code/game/machinery/computer/arcade.dm]: lol no mouse gun in an arcade machine.

Comment (BurgerLUA) [reply code/game/machinery/computer/arcade.dm]: Removed. Didn't know what a mouse gun was until I got it in a gift.

File: code/game/machinery/computer/arcade.dm (hunk 1 above)
Comment (BurgerLUA) [reply code/game/machinery/computer/arcade.dm]: Fair enough.

File: code/game/machinery/computer/arcade.dm [hunk 3]
```diff
@@ -24,11 +23,31 @@
...
+							/obj/item/clothing/mask/fakemoustache			= 11,
+							/obj/item/clothing/mask/gas/clown_hat			= 11,
+							/obj/item/clothing/mask/gas/mime				= 11,
+							/obj/item/weapon/gun/energy/wand/toy			= 11,
+							/obj/item/weapon/stamp/clown					= 11,
+							/obj/item/device/binoculars						= 11,
+							/obj/item/device/megaphone						= 11,
+							/obj/random/booze								= 55,
+							/obj/item/weapon/banhammer						= 11,
+							/obj/item/weapon/sord							= 11,
```
Comment (skull132) [inline code/game/machinery/computer/arcade.dm]: Comma at the end here is failing travis.

File: code/game/objects/items/toys.dm [hunk 4]
```diff
@@ -367,9 +367,13 @@
 		src.add_fingerprint(user)
 		return
 
+/obj/item/toy/sword/New()
```
Comment (skull132) [inline code/game/objects/items/toys.dm]: `Initialize()` instead of new. Make sure to include `. = ..()`

**Comment by LordFowl** [comment]
Renaming toys is a terrible idea.

File: code/game/machinery/computer/arcade.dm [hunk 5]
```diff
@@ -24,11 +23,31 @@
...
+							/obj/random/plushie								= 55,
+							/obj/item/toy/cultsword							= 11,
+							/obj/item/toy/balloon							= 22,
+							/obj/item/toy/syndicateballoon					= 11,
+							/obj/item/toy/nanotrasenballoon					= 11,
+							/obj/item/toy/katana							= 11,
+							/obj/item/toy/bosunwhistle						= 11,
+							/obj/item/weapon/storage/belt/champion			= 11,
+							/obj/item/weapon/pen/invisible					= 11,
+							/obj/item/weapon/grenade/smokebomb				= 1,
```
Comment (Lohikar) [inline code/game/machinery/computer/arcade.dm]: ...

File: code/game/machinery/computer/arcade.dm [hunk 6]
```diff
@@ -24,11 +23,31 @@
...
+							/obj/item/weapon/bikehorn						= 11,
+							/obj/item/weapon/lipstick/random				= 11,
+							/obj/item/clothing/mask/fakemoustache			= 11,
+							/obj/item/clothing/mask/gas/clown_hat			= 11,
+							/obj/item/clothing/mask/gas/mime				= 11,
+							/obj/item/weapon/gun/energy/wand/toy			= 11,
+							/obj/item/weapon/stamp/clown					= 11,
```
Comment (Lohikar) [inline code/game/machinery/computer/arcade.dm]: This literally can't be used by anyone on-station.

File: code/game/machinery/computer/arcade.dm (hunk 3 above)
Comment (Lohikar) [inline code/game/machinery/computer/arcade.dm]: This is literally a sword. Shouldn't be available in this.

File: code/game/objects/items/toys.dm [hunk 7]
```diff
@@ -135,7 +135,7 @@
  * Toy gun: Why isnt this an /obj/item/weapon/gun?
  */
 /obj/item/toy/gun
-	name = "cap gun"
+	name = "revolver"
```
Comment (Lohikar) [inline code/game/objects/items/toys.dm]: No. Just replace the paths to this with the actual cap gun path.

File: code/game/objects/items/toys.dm [hunk 8]
```diff
@@ -217,8 +217,8 @@
  */
 
 /obj/item/toy/crossbow
-	name = "foam dart crossbow"
-	desc = "A weapon favored by many overactive children. Ages 8 and up."
+	name = "mini energy-crossbow"
```
Comment (Lohikar) [inline code/game/objects/items/toys.dm]: N O

File: code/game/objects/items/toys.dm [hunk 9]
```diff
@@ -335,22 +335,22 @@
  * Toy swords
  */
 /obj/item/toy/sword
-	name = "toy sword"
-	desc = "A cheap, plastic replica of an energy sword. Realistic sounds! Ages 8 and up."
+	name = "energy sword"
+	desc = "Upon further examination, this one seems to be a cheap, plastic replica of an energy sword. Realistic sounds and colors! Ages 8 and up."
```
Comment (Lohikar) [inline code/game/objects/items/toys.dm]: Not a fan of this change. Old desc was fine.

File: code/game/objects/items/toys.dm (hunk 8 above)
Comment (BurgerLUA) [reply code/game/objects/items/toys.dm]: why are you fine with everything else being renamed but not the energy crosbow

Comment (Lohikar) [reply code/game/objects/items/toys.dm]: I'm not a fan of them being renamed either, this is just the one I take the most offense to.

**Comment by Fire-and-Glory** [comment]
Renaming toys to traitor weapons sounds like a hideously bad idea.

**Comment by skull132** [comment]
@Lohikar check review please.

**Pull request merged**